### PR TITLE
abseil-cpp: 20200225.2 -> 20200923.2

### PR DESCRIPTION
**Pull request by freezeboy**
###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by andersk** [comment]
Last I checked, upgrading abseil-cpp breaks a number of packages; see #100827. We may need to provide multiple versions.

**Comment by SuperSandro2000** [comment]
Result of `nixpkgs-review pr 107846` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages marked as broken and skipped:</summary>
  <ul>
    <li>google-cloud-cpp</li>
    <li>linuxPackages_4_4.sysdig</li>
    <li>linuxPackages_4_9.sysdig</li>
    <li>linuxPackages_hardkernel_4_14.sysdig</li>
    <li>linuxPackages_hardkernel_latest.sysdig</li>
  </ul>
</details>
<details>
  <summary>13 packages failed to build:</summary>
  <ul>
    <li>linuxPackages-libre.sysdig</li>
    <li>or-tools</li>
    <li>python37Packages.mask-rcnn</li>
    <li>python37Packages.ortools</li>
    <li>python37Packages.tensorflow_2 (python37Packages.tensorflow-build_2)</li>
    <li>python38Packages.mask-rcnn</li>
    <li>python38Packages.ortools</li>
    <li>python38Packages.tensorflow_2 (python38Packages.tensorflow-build_2)</li>
    <li>python39Packages.ortools</li>
    <li>python39Packages.tensorflow_2 (python39Packages.tensorflow-build_2)</li>
    <li>rippled</li>
    <li>rippled-validator-keys-tool</li>
    <li>tts</li>
  </ul>
</details>
<details>
  <summary>21 packages built:</summary>
  <ul>
    <li>abseil-cpp</li>
    <li>bear</li>
    <li>grpc</li>
    <li>linuxPackages.sysdig (linuxPackages_5_4.sysdig)</li>
    <li>linuxPackages_4_14.sysdig</li>
    <li>linuxPackages_4_19.sysdig</li>
    <li>linuxPackages_5_10.sysdig (linuxPackages_latest.sysdig)</li>
    <li>linuxPackages_5_9.sysdig</li>
    <li>linuxPackages_hardened.sysdig</li>
    <li>linuxPackages_latest-libre.sysdig</li>
    <li>linuxPackages_latest_hardened.sysdig</li>
    <li>linuxPackages_latest_xen_dom0.sysdig</li>
    <li>linuxPackages_latest_xen_dom0_hardened.sysdig</li>
    <li>linuxPackages_lqx.sysdig</li>
    <li>linuxPackages_testing_bcachefs.sysdig</li>
    <li>linuxPackages_xen_dom0.sysdig</li>
    <li>linuxPackages_xen_dom0_hardened.sysdig</li>
    <li>linuxPackages_zen.sysdig</li>
    <li>murmur</li>
    <li>qv2ray</li>
    <li>sysdig</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107846` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages marked as broken and skipped:
</summary>
<ul>
<li>python37Packages.mask-rcnn</li>
<li>python38Packages.mask-rcnn</li>
<li>python39Packages.tensorflow-build_2</li>
<li>python39Packages.tensorflow_2</li>
<li>sysdig</li>
</ul>
</details>
<details>

  <summary>2 packages failed to build and already failed to build on hydra master:</summary>
  <ul>
<li>bear: log <a href="https://termbin.com/d3j4v">https://termbin.com/d3j4v</a></li>
<li>qv2ray: log <a href="https://termbin.com/i0rk">https://termbin.com/i0rk</a></li>
  </ul>
</details>
<details>

  <summary>4 packages failed to build and are new build failures:</summary>
  <ul>
<li>python37Packages.tensorflow_2: log was empty</li>
<li>(python37Packages.tensorflow-build_2): log was empty</li>
<li>python38Packages.tensorflow_2: log was empty</li>
<li>(python38Packages.tensorflow-build_2): log was empty</li>
  </ul>
</details>
<details>
<summary>
2 packages built:
</summary>
<ul>
<li>abseil-cpp</li>
<li>grpc</li>
</ul>
</details>

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review which is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 107846` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
7 packages marked as broken and skipped:
</summary>
<ul>
<li>google-cloud-cpp</li>
<li>linuxPackages_4_4.sysdig</li>
<li>linuxPackages_4_9.sysdig</li>
<li>linuxPackages_hardkernel_4_14.sysdig</li>
<li>linuxPackages_hardkernel_latest.sysdig</li>
<li>python39Packages.tensorflow-build_2</li>
<li>python39Packages.tensorflow_2</li>
</ul>
</details>
<details>

  <summary>13 packages failed to build and are new build failures:</summary>
  <ul>
<li>linuxPackages-libre.sysdig: log was empty</li>
<li>or-tools: log <a href="https://termbin.com/5hud">https://termbin.com/5hud</a></li>
<li>python37Packages.mask-rcnn: log was empty</li>
<li>python37Packages.ortools: log <a href="https://termbin.com/rlf5">https://termbin.com/rlf5</a></li>
<li>python37Packages.tensorflow_2: log was empty</li>
<li>(python37Packages.tensorflow-build_2): log was empty</li>
<li>python38Packages.mask-rcnn: log was empty</li>
<li>python38Packages.ortools: log <a href="https://termbin.com/3944">https://termbin.com/3944</a></li>
<li>python38Packages.tensorflow_2: log was empty</li>
<li>(python38Packages.tensorflow-build_2): log was empty</li>
<li>python39Packages.ortools: log <a href="https://termbin.com/4mx3">https://termbin.com/4mx3</a></li>
<li>rippled: log <a href="https://termbin.com/u7bj">https://termbin.com/u7bj</a></li>
<li>rippled-validator-keys-tool: log was empty</li>
  </ul>
</details>
<details>
<summary>
21 packages built:
</summary>
<ul>
<li>abseil-cpp</li>
<li>bear</li>
<li>grpc</li>
<li>linuxPackages.sysdig (linuxPackages_5_4.sysdig)</li>
<li>linuxPackages_4_14.sysdig</li>
<li>linuxPackages_4_19.sysdig</li>
<li>linuxPackages_5_10.sysdig (linuxPackages_latest.sysdig)</li>
<li>linuxPackages_5_9.sysdig</li>
<li>linuxPackages_hardened.sysdig</li>
<li>linuxPackages_latest-libre.sysdig</li>
<li>linuxPackages_latest_hardened.sysdig</li>
<li>linuxPackages_latest_xen_dom0.sysdig</li>
<li>linuxPackages_latest_xen_dom0_hardened.sysdig</li>
<li>linuxPackages_lqx.sysdig</li>
<li>linuxPackages_testing_bcachefs.sysdig</li>
<li>linuxPackages_xen_dom0.sysdig</li>
<li>linuxPackages_xen_dom0_hardened.sysdig</li>
<li>linuxPackages_zen.sysdig</li>
<li>murmur</li>
<li>qv2ray</li>
<li>sysdig</li>
</ul>
</details>

**Comment by marsam** [comment]
addressed by https://github.com/NixOS/nixpkgs/pull/112951
